setup.py

```diff
@@ -0,0 +1,109 @@
+import os
+import re
+import shutil
+from pathlib import Path
+
+from setuptools import setup, find_packages
+
+# Copy src/run_tribler.py --> src/tribler/run.py to make it accessible in entry_points scripts.
+shutil.copy("src/run_tribler.py", "src/tribler/run.py")
+
+
+def read_version_from_file(file_path):
+    with open(file_path, "r", encoding="utf-8") as file:
+        file_content = file.read()
+    # Use regular expression to find the version pattern
+    version_match = re.search(r"^version_id = ['\"]([^'\"]*)['\"]", file_content, re.M)
+    if version_match:
+        version_str = version_match.group(1)
+        return version_str.split("-")[0]
+    raise RuntimeError("Unable to find version string.")
+
+
+version_file = os.path.join('src', 'tribler', 'core', 'version.py')
+version = read_version_from_file(version_file)
+
+
+setup(
+    name="Tribler",
+    version=version,
+    description="Privacy enhanced BitTorrent client with P2P content discovery",
+    long_description=Path('README.rst').read_text(encoding="utf-8"),
+    long_description_content_type="text/x-rst",
+    author="Tribler Team",
+    author_email="info@tribler.org",
+    url="https://www.tribler.org",
+    packages=find_packages(where="src"),
+    package_dir={"": "src"},
+    include_package_data=True,
+    install_requires=[
+        "aiohttp==3.9.0",
+        "aiohttp-apispec==2.2.3",
+        "anyio==3.7.1",
+        "chardet==5.1.0",
+        "configobj==5.0.8",
+        "cryptography==42.0.5",
+        "Faker==18.11.2",
+        "libnacl==1.8.0",
+        "lz4==4.3.2",
+        "marshmallow==3.19.0",
+        "networkx==3.1",
+        "pony==0.7.17",
+        "psutil==5.9.5",
+        "pydantic==1.10.11",
+        "PyOpenSSL==24.0.0",
+        "pyyaml==6.0",
+        "sentry-sdk==1.31.0",
+        "yappi==1.4.0",
+        "yarl==1.9.2",
+        "bitarray==2.7.6",
+        "pyipv8==2.13.0",
+        "libtorrent==1.2.19",
+        "file-read-backwards==3.0.0",
+        "Brotli==1.0.9",
+        "human-readable==1.3.2",
+        "colorlog==6.7.0",
+        "filelock==3.13.0",
+        "ipv8-rust-tunnels==0.1.17",
+        "Pillow==10.2.0",
+        "PyQt5==5.15.1",
+        "PyQt5-sip==12.8.1",
+        "pyqtgraph==0.12.3",
+        "PyQtWebEngine==5.15.2",
+        "setuptools==65.5.1; sys_platform == 'darwin'",
+        "text-unidecode==1.3; sys_platform == 'darwin'",
+        "defusedxml==0.7.1; sys_platform == 'linux2' or sys_platform == 'linux'",
+        "markupsafe==2.0.1; sys_platform == 'linux2' or sys_platform == 'linux'",
+        "PyGObject==3.44.1; sys_platform == 'linux2' or sys_platform == 'linux'",
+        "requests==2.31.0",
+    ],
```

Comment:
While I agree that it is usually recommended to keep `setup.py` and `requirements.txt` separate and independent to provide flexibility and to avoid over-specification of dependencies, it is more important for libraries. As Tribler is an application and we want to ensure environment consistency, it is less critical for Tribler to have a separate list of requirements in `setup.py` and `requirements.txt`. So I'd prefer to parse `install_requirements` from `requirements.txt` using a function like:

```python
def read_requirements(file_name, directory='.'):
    file_path = os.path.join(directory, file_name)
    requirements = []
    with open(file_path, 'r', encoding='utf-8') as file:
        for line in file:
            # Check for a nested requirements file
            if line.startswith('-r'):
                nested_file = line.split(' ')[1].strip()
                requirements += read_requirements(nested_file, directory)
            elif not line.startswith('#') and line.strip() != '':
                requirements.append(line.strip())
    return requirements

base_dir = os.path.dirname(os.path.abspath(__file__))

install_requires = read_requirements('requirements-build.txt', base_dir)
extras_require = {
    'dev': read_requirements('requirements-test.txt', base_dir),
}

setup(
    ...
    install_requires=install_requires,
    extras_require=extras_require,
    ...
)
```

But while I prefer this method to avoid duplication, I'm OK with the current code as well, because, as I said, keeping `requirements.txt` and `install_requires` separate is considered normal practice.

Reply:
I have added the requested change.